lib/puppet/provider/package/gem.rb

```diff
@@ -121,13 +127,18 @@ def install(useversion = true)
           command << source
         else
           # interpret it as a gem repository
-          command << "--source" << "#{source}" << resource[:name]
+          command << "--source" << "#{source}"
         end
       end
+    end
+
+    if Puppet::Util::Package.versioncmp(gem_version(command(:gemcmd)), '2.0.0') == -1
```

Comment:
Does `versioncmp` always return `-1`, `0`, `1` or can it return a more negative version? Guess I'm more used to seeing `versioncmp(a, b) < 0`, or `== 0`, or `> 0`

Reply:
Yes: 

https://github.com/puppetlabs/puppet/blob/4.10.x/lib/puppet/functions/versioncmp.rb#L15

Chose that to match this: 

https://github.com/puppetlabs/puppet/blob/4.10.x/lib/puppet/provider/package/pip.rb#L81